cloud-docs: Update HCP TF security model (#803)
hashicorp/terraform-docs-common: website/docs/cloud-docs/architectural-details/security-model.mdx

The logs from a Terraform plan or apply operation are visible to any user with at least “read” level access in the associated workspace. While Terraform tries to avoid writing sensitive information to logs, redactions are best-effort. This feature should not be treated as a security boundary, but instead as a mechanism to mitigate accidental exposure. Additionally, HCP Terraform is unable to protect against malicious users who attempt to use Terraform logs to exfiltrate sensitive data.

> ### Redaction of ephemeral values in Terraform logs

Reviewer: ```suggestion
### Redact ephemeral values from Terraform logs
```

> The logs from a Terraform plan or apply operation are visible to any user with at least “read” level access in the associated workspace. Terraform tries to avoid writing ephemeral values to logs

Reviewer: Suggest adding a link to ephemeral values so folks can learn more:
```suggestion
The logs from a Terraform plan or apply operation are visible to any workspace's users with **Read** permissions. Terraform attempts to avoid writing [ephemeral values](/terraform/language/resources/ephemeral) to logs, but Terraform cannot guarantee that all providers will not log ephemeral values.
```

> and provider developers are discouraged from logging those as well but Terraform cannot provide guarantees that providers will not log ephemeral values.

Reviewer: If you like the suggestion above, this is just a reminder to remove this line:
```suggestion

```

> You can reduce the risk by only [using trusted modules and providers within Terraform configuration](#malicious-terraform-providers-or-modules).

Reviewer: Can you specify what they are reducing the risk of?

Author: They are reducing the risk of ephemeral values being logged by providers by reducing the chances of using malicious providers that would attempt to do so intentionally (rather than accidentally).

> ### Redaction of ephemeral values in memory

Reviewer: ```suggestion
### Redact ephemeral values in memory
```

> Ephemeral Values prevents values from being persisted to disk (as part of a plan file or state file) but no efforts are made to protect ephemeral values from memory analysis of the running application.

Reviewer: Is there any recommendation we can give to folks here? Is this something they need to worry about? Or more like something we are giving them a heads-up about?
```suggestion
Terraform does not persist ephemeral values to plan or state files. However, Terraform does not protect ephemeral values from a memory analysis of your running application.
```

Author: I'm fine with the proposed first part but I think the last part could be a bit confusing (and I mean both before and after the proposed change 😄 ).

What we mean by "running application" here is Terraform. So I suppose we could either omit the part entirely or say "memory analysis of running Terraform"?

Reviewer: Oooh love it! How about this then?
```suggestion
Terraform does not persist ephemeral values to plan or state files. However, Terraform does not protect ephemeral values from a memory analysis of Terraform while its running.
```
